compiler/passes/convert-typed-uast.cpp

```diff
@@ -2252,11 +2263,30 @@ struct ConvertTypeHelper {
         continue;
       }
       Type* ft = tc_->convertType(qt.type());
+
+      // NOTE: During the intermediate stage of development of this conversion
+      // pass, we need to preserve field initialization expressions in case
+      // they are used in production's resolution later.
+      //
+      // For example, strings are used by both the compiler and the typed
+      // converter.
+      auto declID = rf.fieldDeclId(i);
+      const AstNode* declAst = parsing::idToAst(tc_->context, declID);
+      const VarLikeDecl* field = declAst->toVarLikeDecl();
+      if (!field) {
+        INT_ASSERT(declAst->isForwardingDecl());
+        field = declAst->toForwardingDecl()->expr()->toVarLikeDecl();
+      }
+      Expr* initExpr = nullptr;
+      if (field->initExpression()) {
+        initExpr = tc_->untypedConverter->convertAST(field->initExpression());
+      }
+
       UniqueString name = rf.fieldName(i);
       VarSymbol* v = new VarSymbol(astr(name), ft);
       v->qual = tc_->convertQualifier(qt.kind());
       v->makeField();
-      at->fields.insertAtTail(new DefExpr(v));
+      at->fields.insertAtTail(new DefExpr(v, initExpr));
```

Comment:
Actually, I'm confused why we ever could avoid inserting an initialization expression. Those aren't types, so they could persist into the final executable. I see that we're doing an untyped conversion here (presumably, later it will be a typed conversion), but I can't imagine not converting the `initExpr` at all. What am I missing?

Reply:
that initialization expression is actually untyped, and is left in for the production resolver to deal with so that non-type-converted code (i.e. some essential internal module stuff) can use those same types.